src/librustc/ich/impls_ty.rs

```diff
@@ -686,7 +686,7 @@ impl<'a> HashStable<StableHashingContext<'a>> for ty::Generics {
     }
 }
 
-impl_stable_hash_for!(enum ty::GenericParameterDef {
+impl_stable_hash_for!(enum ty::GenericParam {
```

Comment:
Hmm, I know I agreed to the param vs arg split, but I have to say that *overall* these new names feel a bit less clear. :/

Reply:
I guess naming isn't one of the two hardest problems in Computer Science for nothing 😄
Maybe this should be one of the agenda items for the next rustc team meeting ;)